### PR TITLE
ci(teamcity): Extract projects into its own kotlin files

### PR DESCRIPTION
**Pull request by scinos**
### Background

This PR is the first of a series of refactors I'd like to do to TeamCity project structure to have better names and more clarity. Right now we have the structure

```
Calypso
	- Build base images
	- Build docker image
	- Check code style
	- Check code style for branches
	- Run browser e2e tests (desktop)
	- Run unit tests
	- Desktop app
		- Run e2e tests
	- WPCom plugins
		- Editor ToolKit
```

As you can see, it doesn't have a very good hierarchy with some mixed concerns. I'd like to move to 

```
Calypso
	- Base docker images
	- Code style
	- Web app
		- Code style
		- Unit tests
		- Docker image
		- E2E tests (browser)
	- Desktop app
		- E2E tests
	- WPCom plugins
		- Editor ToolKit
```

Using the following principles:
- Group related builds into sub projects: `Web App`, `Desktop App` and `WPCom plugins`. For example, in the future we may have `Packages` as well.
- Drop verbs from the name (run, build, check...), as they are redundant
- "Root" builds (those hanging on the main project) are builds not specific to an app, but related to the whole repo. Right now we will have `Code style` (run `eslint` across the whole repo) and `Base docker images` (build docker images used to run other builds). An alternative would be to have a sub project named `Operations` or something like that (I wasn't able to find a good name tbh)

### Changes proposed in this Pull Request

**Move `Project` definitions to their own files**

I think it is a good idea to have `BuildType`s and `Project` in the same file, as we always have a 1:N relationship (a `Project` can have many `BuildType`s, but a `BuildType` only belongs to one `Project`) and this structure reinforces that relationship. Plus it is easy to see which `params` a `BuildType` is using in case they have been defined in the `Project`.

**Add `id()` fields to the moved builds**

This allows TeamCity to match the build with an existing build history. This gives us freedom to change the object name as we see fit.

### Testing instructions

Verify TeamCity tests still work and it doesn't complain about a malformed DSL.

### CODE REVIEW

**Comment by scinos** [comment]
After a few experiments, I finally found the pattern on the project and build ids. I'm writing my findings here for future reference:

- If a `Project`/`BuildType` doesn't have an explicit `id()`, its id is computed based on the object name.
- Subprojects must not concatenate the parent project id. For example, TeamCity UI shows id `calypso_WPComPlugins` for WPCom project, but the DSL must have just `id("WPComPlugins")` (in other words, TeamCity will concatenate the id of the parent project automatically.
- Builds must include the project id, but not the parent project id. For example, TeamCity UI shows `calypso_WPComPlugins_EditorToolKit` for the only build inside WPCom project, but the DSL should say `id("WPComPlugins_EditorToolKit")`. In other words, TC will _not_ concatenate the project id, but will concatenate the parent project id automatically. Go figure.

File: .teamcity/_self/projects/DesktopApp.kt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+import _self.bashNodeScript
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildStep
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import jetbrains.buildServer.configs.kotlin.v2019_2.ParameterDisplay
+import jetbrains.buildServer.configs.kotlin.v2019_2.Project
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.*
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.vcs
+
+object DesktopApp : Project({
```
Comment (scinos) [inline .teamcity/_self/projects/DesktopApp.kt:11]: This is a copy/paste from the same project in `settings.kts`, plus the change of `id("WpDesktop")`

File: .teamcity/_self/projects/DesktopApp.kt [hunk 2]
```diff
@@ -0,0 +1,166 @@
+package _self.projects
+
+import _self.bashNodeScript
```
Comment (scinos) [inline .teamcity/_self/projects/DesktopApp.kt:3]: This is a copy/paste from the same project in settings.kts, plus the addition of id("WpDesktop") and `id("WpDesktop_DesktopE2ETests")` so we can rename these objects.

File: .teamcity/_self/projects/WPComPlugins.kt [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
 import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import jetbrains.buildServer.configs.kotlin.v2019_2.Project
 import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.PullRequests
 import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.commitStatusPublisher
 import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.pullRequests
 import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.vcs
 
-object WPComPlugins_EditorToolKit : BuildType({
-	name = "Editor ToolKit"
+object WPComPlugins : Project({
```
Comment (scinos) [inline .teamcity/_self/projects/WPComPlugins.kt:11]: This is a copy/paste from the same project in settings.kts, plus the addition of `id("WPComPlugins")`

File: .teamcity/settings.kts [hunk 4]
```diff
@@ -801,165 +801,3 @@ object WpCalypso : GitVcsRoot({
 	}
 })
 
-object WpDesktop : Project({
-	name = "Desktop app"
-	buildType(WpDesktop_DesktopE2ETests)
-
-	params {
-		text("docker_image_dekstop", "registry.a8c.com/calypso/ci-desktop:latest", label = "Docker image", description = "Docker image to use for the run", allowEmpty = true)
```
Comment (scinos) [inline .teamcity/settings.kts:809]: Note this typo (`dekstop`) has been fixed in `.teamcity/_self/projects/DesktopApp.kt`

**Review by worldomonation — APPROVED** [review]
The new structure makes more sense. Thank you for this.

**Review by noahtallen — APPROVED** [review]
Fantastic, this makes a lot of sense to me.

**Pull request merged**